--- a/src/Eplists.pm
+++ b/src/Eplists.pm
@@ -1064,7 +1064,7 @@
     $OutputData{Data}[$i] = $Config{Format};
   }
 
-  foreach my $key ( keys %OutputData ) {
+  foreach my $key ( 'Prefix', 'Data' ) { 
     for (my $i = 0; $i < scalar @{$OutputData{$key}}; $i++) {
       my %FormatM = ( "s" => 2, "e" => 2, "n" => 3 );
       foreach("s", "e", "n") {
--- a/src/EplistsClient.pm
+++ b/src/EplistsClient.pm
@@ -315,7 +315,8 @@
   }
 
   sub startssl {
-    IO::Socket::SSL->start_SSL( $Socket );
+      IO::Socket::SSL->start_SSL( $Socket, 
+      SSL_verify_mode => NONE) or die $SSL_ERROR;
   }
 
   ##########################
--- a/src/svdrpsend-ng.pl
+++ b/src/svdrpsend-ng.pl
@@ -123,7 +123,7 @@
                                 UseSSL		=> $Config{UseSSL}
                         },
                         $Cmd
-                        ) || exit 1;
+                        ) && exit 1;
                                                                                                                                                                                                                                                                                 
 
 __END__
