Index: vdr-plugin-audiorecorder-2.0.0rc5+git20141109/Makefile
===================================================================
--- vdr-plugin-audiorecorder-2.0.0rc5+git20141109.orig/Makefile	2017-07-09 19:34:20.846139000 +0200
+++ vdr-plugin-audiorecorder-2.0.0rc5+git20141109/Makefile	2017-07-09 19:34:20.842138980 +0200
@@ -87,7 +87,7 @@
 
 ### The object files (add further files here):
 
-OBJS = $(PLUGIN).o mainmenu.o browse.o browse-item.o column.o dispatcher.o audioreceiver.o postdata.o trackinfo.o postproc.o rds.o convert.o cache.o xml-cache.o xml-base.o setup.o mpa-frame.o
+OBJS = $(PLUGIN).o mainmenu.o browse.o browse-item.o column.o dispatcher.o audioreceiver.o postdata.o trackinfo.o postproc.o rds.o convert.o cache.o xml-cache.o xml-base.o setup.o mpa-frame.o tinyxml/tinyxml.o tinyxml/tinyxmlerror.o tinyxml/tinyxmlparser.o
 
 ### The main target:
 
Index: vdr-plugin-audiorecorder-2.0.0rc5+git20141109/setup.c
===================================================================
--- vdr-plugin-audiorecorder-2.0.0rc5+git20141109.orig/setup.c	2017-07-09 19:34:20.846139000 +0200
+++ vdr-plugin-audiorecorder-2.0.0rc5+git20141109/setup.c	2017-07-09 19:34:20.842138980 +0200
@@ -15,12 +15,12 @@
         1,  /* pause postprocessing if osd is opened */
         8,  /* max. tracks in postprocessing queue */
         1,  /* fade in mode */
-        9,  /* fade in seconds */
-        2,  /* fade out mode */
-        12, /* fade out seconds */
+        1,  /* fade in seconds */
+        1,  /* fade out mode */
+        1,  /* fade out seconds */
         1,  /* audio codec */
         NUM_CODECS,  /* number of audio codecs */
-        9,  /* bitrate */
+        12, /* bitrate */
         0   /* file_pattern */
 };
 
