Index: lcdproc-0.5.9/LCDd.conf
===================================================================
--- lcdproc-0.5.9.orig/LCDd.conf
+++ lcdproc-0.5.9/LCDd.conf
@@ -34,7 +34,7 @@
 #            the driver modules and will thus not be able to
 #            function properly.
 # NOTE: Always place a slash as last character !
-DriverPath=server/drivers/
+DriverPath=usr/lib/x86_64-linux-gnu/lcdproc
 
 # Tells the server to load the given drivers. Multiple lines can be given.
 # The name of the driver is case sensitive and determines the section
