Index: vdr-plugin-imonlcd-1.0.1+git20150204/Makefile
===================================================================
--- vdr-plugin-imonlcd-1.0.1+git20150204.orig/Makefile
+++ vdr-plugin-imonlcd-1.0.1+git20150204/Makefile
@@ -55,8 +55,8 @@ SOFILE = libvdr-$(PLUGIN).so
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += $(shell freetype-config --cflags)
-LIBS += $(shell freetype-config --libs)
+INCLUDES += $(shell pkg-config --cflags freetype2)
+LIBS += $(shell pkg-config --libs freetype2)
 
 DEFINES += -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 
