From d4e1b2c241cd70710218c20d61043407a4a003c5 Mon Sep 17 00:00:00 2001
From: wsnipex <wsnipex@a1.net>
Date: Tue, 7 Jul 2015 10:04:56 +0200
Subject: [PATCH] allow to read airport key from /etc/shairplay/airport.key

---
 src/shairplay.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/shairplay.c b/src/shairplay.c
index 4b873f1..94f3493 100644
--- a/src/shairplay.c
+++ b/src/shairplay.c
@@ -348,9 +348,11 @@ main(int argc, char *argv[])
 
        raop = raop_init_from_keyfile(10, &raop_cbs, "airport.key", NULL);
        if (raop == NULL) {
-               fprintf(stderr, "Could not initialize the RAOP service\n");
-               fprintf(stderr, "Please make sure the airport.key file is in the current directory.\n");
-               return -1;
+               raop = raop_init_from_keyfile(10, &raop_cbs, "/etc/shairplay/airport.key", NULL);
+               if (raop == NULL) {
+                       fprintf(stderr, "Could not initialize the RAOP service\n");
+                       return -1;
+               }
        }
 
        if (strlen(options.password)) {
-- 
1.9.1

