Index: vdr-plugin-softhddevice-openglosd-0.6.1rc1-25-569fde5/Makefile
===================================================================
--- vdr-plugin-softhddevice-openglosd-0.6.1rc1-25-569fde5.orig/Makefile	2017-02-07 01:14:59.749046627 +0100
+++ vdr-plugin-softhddevice-openglosd-0.6.1rc1-25-569fde5/Makefile	2017-02-07 01:14:59.749046627 +0100
@@ -37,7 +37,7 @@
 
 CONFIG += -DAV_INFO -DAV_INFO_TIME=3000	# info/debug a/v sync
 CONFIG += -DUSE_PIP			# PIP support
-#CONFIG += -DHAVE_PTHREAD_NAME		# supports new pthread_setname_np
+CONFIG += -DHAVE_PTHREAD_NAME		# supports new pthread_setname_np
 #CONFIG += -DNO_TS_AUDIO		# disable ts audio parser
 #CONFIG += -DUSE_TS_VIDEO		# build new ts video parser
 CONFIG += -DUSE_MPEG_COMPLETE		# support only complete mpeg packets
