Index: vdr-plugin-radio-1.0.0/radioskin.c
===================================================================
--- vdr-plugin-radio-1.0.0.orig/radioskin.c
+++ vdr-plugin-radio-1.0.0/radioskin.c
@@ -14,10 +14,10 @@ const cRadioSkin radioSkin[eRadioSkinMax
   {
     // dummy
     "dummy",
-    0xFFFCFCFC, // clrTitleBack
-    0xFF000000, // clrTitleText
+    0xB8003c78, // clrTitleBack
+    0xFFFFFFFF, // clrTitleText
     0x7F000000, // clrBack
-    0xFFFCFCFC, // clrText
+    0xB8FFFFDC, // clrText
   },
   {
     // Classic
