Description: Remove unneeded sys.path call (potential security issue)
Author: C de-Avillez <hggdh2@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/795159
Bug-Debian: http://bugs.debian.org/638332
Last-Update: 2011-08-19

--- a/ajaxterm.py
+++ b/ajaxterm.py
@@ -18,8 +18,6 @@
 	sha1 = sha.new
 
 os.chdir(os.path.normpath(os.path.dirname(__file__)))
-# Optional: Add QWeb in sys path
-sys.path[0:0]=glob.glob('../../python')
 
 import qweb, codecs
 utf8decoder = codecs.getincrementaldecoder('utf8')()
