Index: vdr-2.2.0/osdbase.c
===================================================================
--- vdr-2.2.0.orig/osdbase.c	2017-11-22 21:41:14.478868460 +0100
+++ vdr-2.2.0/osdbase.c	2017-11-22 21:41:14.470868421 +0100
@@ -522,7 +522,7 @@
       const char *s = item->Text();
       i = 0;
       item_nr = 0;
-      if (s && (s = skipspace(s)) != '\0' && '0' <= s[i] && s[i] <= '9') {
+      if (s && (s = skipspace(s)) != NULL && '0' <= s[i] && s[i] <= '9') {
          do {
             item_nr = item_nr * 10 + (s[i] - '0');
             }
