Index: vdr-plugin-externalplayer-0.3.2/externalplayer-config.c
===================================================================
--- vdr-plugin-externalplayer-0.3.2.orig/externalplayer-config.c	2013-04-21 18:06:22.000000000 +0200
+++ vdr-plugin-externalplayer-0.3.2/externalplayer-config.c	2017-07-02 18:41:04.862972603 +0200
@@ -131,9 +131,10 @@
 
 string cExternalplayerConfig::ReadConfigFile(const string &filename) {
   ifstream playerConfigStream;
-  playerConfigStream.open(filename.c_str(), ios::in);
-
-  if (playerConfigStream == NULL) {
+  try {
+    playerConfigStream.open(filename.c_str(), ios::in);
+  }
+  catch (ifstream::failure e) {
     throw FileNotFoundException(filename);
   }
 
